src/tests/pools/D3M4626TypePool.t.sol

```diff
@@ -0,0 +1,168 @@
+
+contract ERC4626 is ERC4626Abstract {
+    constructor(
+        ERC20 _asset,
+        string memory _name,
+        string memory _symbol
+    ) ERC4626Abstract(_asset, _name, _symbol) {}
+
+    function totalAssets() public view virtual override returns (uint256) {
+        return asset.balanceOf(address(this));
+    }
+}
+
+contract D3M4626TypePoolTest is D3MPoolBaseTest {
+    
+    D3M4626TypePool pool;
+    ERC4626 vault;
+    bytes32 constant ILK = "TEST-ILK";
+
+    function setUp() public {
+        baseInit("D3M4626TypePool");
+
+        vault = new ERC4626(ERC20(address(dai)), "dai vault", "DV");
+
+        setPoolContract(pool = new D3M4626TypePool(ILK, address(hub), address(dai), address(vault)));
+
+        dai.approve(address(vault), type(uint256).max);
+    }
+
+    function invariant_dai_value() public {
+        assertEq(address(pool.dai()), address(dai));
+    }
+
+    function invariant_vault_value() public {
+        assertEq(address(pool.vault()), address(vault));
+    }
+
+    function invariant_vat_value() public {
+        assertEq(address(pool.vat()), address(vat));
+    }
+
+    function invariant_ilk_value() public {
+        assertEq(pool.ilk(), ILK);
+    }
+
+    function test_cannot_file_hub_no_auth() public {
+        pool.deny(address(this));
+        vm.expectRevert("D3M4626TypePool/not-authorized");
+        pool.file("hub", address(123));
+    }
+
+    function test_deposit_calls_vault_deposit() public {
+        deal(address(dai), address(pool), 1);
+        vm.prank(address(hub)); pool.deposit(1);
+        
+        assertEq(pool.assetBalance(), 1);
+        assertEq(dai.balanceOf(address(pool)), 0);
+    }
+
+    function test_withdraw_calls_vault_withdraw() public {
+        deal(address(dai), address(pool), 1);
+        vm.prank(address(hub)); pool.deposit(1);
+        
+        vm.prank(address(hub)); pool.withdraw(1);
+        
+        assertEq(pool.assetBalance(), 0);
+        assertEq(dai.balanceOf(address(hub)), 1);
+    }
+
+    function test_withdraw_calls_vault_withdraw_vat_caged() public {
+        deal(address(dai), address(pool), 1);
+        vm.prank(address(hub)); pool.deposit(1);
+        
+        vat.cage();
+        vm.prank(address(hub)); pool.withdraw(1);
+
+        assertEq(pool.assetBalance(), 0);
+        assertEq(dai.balanceOf(address(hub)), 1);
+    }
+
+    function test_redeemable_returns_vault() public {
+        assertEq(pool.redeemable(), address(vault));
+    }
+
+    function test_exit_vault_tokens() public {
+        deal(address(dai), address(this), 1e18);
+        vault.deposit(1e18, address(this));
+        uint256 tokens = vault.totalSupply();
+        vault.transfer(address(pool), tokens);
+        assertEq(vault.balanceOf(address(this)), 0);
+        assertEq(vault.balanceOf(address(pool)), tokens);
+
+        end.setArt(tokens);
+        vm.prank(address(hub)); pool.exit(address(this), tokens);
```

Comment:
Semantically speaking we should be using the assets amount here, which is 1e18. As this is a super basic test they are the same amount. I think the compound test should be a better one to take as an example for this one.

Reply:
Re-opening so @sunbreak1211 can confirm this is complete.

Reply:
I think this is resolved? @sunbreak1211